scipy/spatial/spherical_voronoi.py

```diff
@@ -0,0 +1,416 @@
+    planar_polygon_surface_area = poly_area(list_vertices)
+    return planar_polygon_surface_area
+
+def calculate_and_sum_up_inner_sphere_surface_angles_Voronoi_polygon(array_ordered_Voronoi_polygon_vertices,sphere_radius):
+    '''Takes an array of ordered Voronoi polygon vertices (for a single generator) and calculates the sum of the inner angles on the sphere surface. The resulting value is theta in the equation provided here: http://mathworld.wolfram.com/SphericalPolygon.html '''
+    num_vertices_in_Voronoi_polygon = array_ordered_Voronoi_polygon_vertices.shape[0] #the number of rows == number of vertices in polygon
+    spherical_polar_polygon_vertices = convert_cartesian_array_to_spherical_array(array_ordered_Voronoi_polygon_vertices)
+    min_vertex_radius = spherical_polar_polygon_vertices[...,0].min()
+    assert sphere_radius - min_vertex_radius < 0.1, "The minimum projected Voronoi vertex r value should match the sphere_radius of {sphere_radius}, but got {r_min}.".format(sphere_radius=sphere_radius,r_min=min_vertex_radius)
+    #two edges (great circle arcs actually) per vertex are needed to calculate tangent vectors / inner angle at that vertex
+    current_vertex_index = 0
+    list_Voronoi_poygon_angles_radians = []
+    while current_vertex_index < num_vertices_in_Voronoi_polygon:
+        current_vertex_coordinate = array_ordered_Voronoi_polygon_vertices[current_vertex_index]
+        if current_vertex_index == 0:
+            previous_vertex_index = num_vertices_in_Voronoi_polygon - 1
+        else:
+            previous_vertex_index = current_vertex_index - 1
+        if current_vertex_index == num_vertices_in_Voronoi_polygon - 1:
+            next_vertex_index = 0
+        else:
+            next_vertex_index = current_vertex_index + 1
+        #try using the law of cosines to produce the angle at the current vertex (basically using a subtriangle, which is a common strategy anyway)
+        current_vertex = array_ordered_Voronoi_polygon_vertices[current_vertex_index] 
+        previous_vertex = array_ordered_Voronoi_polygon_vertices[previous_vertex_index]
+        next_vertex = array_ordered_Voronoi_polygon_vertices[next_vertex_index] 
+        a = calculate_Vincenty_distance_between_spherical_points(current_vertex,next_vertex,sphere_radius)
+        b = calculate_Vincenty_distance_between_spherical_points(next_vertex,previous_vertex,sphere_radius)
+        c = calculate_Vincenty_distance_between_spherical_points(previous_vertex,current_vertex,sphere_radius)
+        pre_acos_term = (math.cos(b) - math.cos(a)*math.cos(c)) / (math.sin(a)*math.sin(c))
+        current_vertex_inner_angle_on_sphere_surface = math.acos(pre_acos_term)
+        list_Voronoi_poygon_angles_radians.append(current_vertex_inner_angle_on_sphere_surface)
+        current_vertex_index += 1
+
+    if abs(pre_acos_term) > 1.0:
+        theta = 0
+    else:
+        theta = np.sum(np.array(list_Voronoi_poygon_angles_radians))
```

Comment:
The call to `np.array()` here is superfluous; `np.sum` will handle lists just fine.